### ISSUE TITLE
add back validation (maybe?) for `disable registry url validation in prune`

### ISSUE THREAD

**Issue by deads2k** [openshift/origin#15291]
This was removed in `disable registry url validation in prune`

reenable 

```
if _, err := url.Parse(o.RegistryUrlOverride); 
   err != nil { in pkg/cmd/admin/prune/images.go
```

**Comment by mfojtik** [comment]
@deads2k @miminar ideally I would drop that option in favor of https://trello.com/c/2cHkrqIu/954-provide-public-pull-url-for-images

**Comment by deads2k** [comment]
> @deads2k @miminar ideally I would drop that option in favor of https://trello.com/c/2cHkrqIu/954-provide-public-pull-url-for-images

Your call.  Presumably the code you're calling will do something reasonable.

**Comment by miminar** [comment]
We need to support the flag anyway. Even the best algorithm we can make cannot be right all the time. 

Therefor the validation should be re-enabled. The argument isn't really a URL, it's rather a shortened pull-spec (just the registry part). I thought we could use [docker's validation](https://github.com/miminar/docker/blob/d192db0d9350222d2b8bb6eba8525b04c3be7d61/registry/config.go#L331), but they still use `url.Parse()`. And their [validation for insecure registries](https://github.com/miminar/docker/blob/d192db0d9350222d2b8bb6eba8525b04c3be7d61/registry/config.go#L370) is private. I'll come up with something.

**Comment by miminar** [comment]
Addressed by #14914
